x/stakeibc/keeper/icacallbacks_detokenize.go

```diff
@@ -21,7 +27,28 @@ func DetokenizeCallback(k Keeper, ctx sdk.Context, packet channeltypes.Packet, a
 	chainId := detokenizeCallback.Deposit.ChainId
 	k.Logger(ctx).Info(utils.LogICACallbackWithHostZone(chainId, ICACallbackID_Detokenize, "Starting detokenize callback"))
 
-	// TODO [LSM]
+	// No action is necessary on a timeout
+	if ackResponse.Status == icacallbackstypes.AckResponseStatus_TIMEOUT {
+		k.Logger(ctx).Error(utils.LogICACallbackStatusWithHostZone(chainId, ICACallbackID_Detokenize,
+			icacallbackstypes.AckResponseStatus_TIMEOUT, packet))
+		return nil
+	}
+
+	// If the iCA failed, update the deposit status
+	if ackResponse.Status == icacallbackstypes.AckResponseStatus_FAILURE {
+		k.Logger(ctx).Error(utils.LogICACallbackStatusWithHostZone(chainId, ICACallbackID_Detokenize,
+			icacallbackstypes.AckResponseStatus_FAILURE, packet))
+
+		k.UpdateLSMTokenDepositStatus(ctx, *detokenizeCallback.Deposit, types.DETOKENIZATION_FAILED)
```

Comment:
I see an example below where the Deposit field for this type wants the dereferenced form, so I think I understand what is going on now although I would have thought that it would require parentheses.  Nevermind about the original comment

Reply:
Yes I understood that part of it in terms of the mechanics because it is similar to C, but I wasn't getting at first that the struct is defined with a pointer (in fact this is something I still want to ask about because a normal Marshal would save the pointer instead of the data which I assume is not useful) but I was confused as much about the precedence and was reading it wrong.  The dereference applies to the whole statement, ie. the dot for a subfield takes precedence over the * for dereferencing.  That's why I was thinking it would need parenthesis but it forced me to go read up a bit on go... (I think it also works this way in C, I just hadn't done it in a few years).  But thank you for carefully explaining all the same.